core/vm/logger.go

```diff
@@ -275,6 +276,27 @@ func (l *StructLogger) CaptureState(pc uint64, op OpCode, gas, cost uint64, scop
 		extraData.Caller = append(extraData.Caller, getWrappedAccountForAddr(l, scope.Contract.Address()))
 	}
 
+	// in reality it is impossible for CREATE to trigger ErrContractAddressCollision
+	if op == CREATE2 && opErr == nil {
```

Comment:
Maybe we can omit `OpErr` since in `CaptureState` it seems always true. And just indicate the executing results of **previous** op.

And currently if the creation is success the data of callee (created account) would be recorded. So we would have duplicated record for a successful CREATE2 (but this should be no harm besides some data?)

Reply:
1. no.  OpErr can be err if stack underflow or out of gas happens.
2. I only record it if collision happens? which will not goes into CaptureEnter, so there will be no  duplicated record?